django/conf/__init__.py

```diff
@@ -105,6 +105,21 @@ def __init__(self, settings_module):
                 setattr(self, setting, setting_value)
                 self._explicit_settings.add(setting)
 
+        if self.SECRET_KEY_FILE and not self.SECRET_KEY:
```

Comment:
I think having `SECRET_KEY` win is probably a reasonable choice here. The directory shouldn't be writable, but I'd hate to see an attack where an attacker's `.secret_key` file was written into the dir and then suddenly magically worked.